### ISSUE TITLE
Incorrect display of macerator recipes in JEI, REI

### ISSUE THREAD

**Issue by Exzept1on** [GregTechCEu/GregTechCEu-1.19#591]
### GregTech CEu Version

Forge - 1.20.1-1.0.16

### Recipe Viewer Installed

REI

### Environment

Singleplayer, Open to LAN, Dedicated Server

### Cross-Mod Interaction

Unsure

### Expected Behavior

When pressing U on any terracotta, I want to see the macerator button at the top of the tabs, because I get wood dust from terracotta. (KubeJS recipe for test)

### Actual Behavior

When pressing U on any terracotta, I don't see the recipe in macerator, although it is added.

### Steps to Reproduce

1. Install GTCEu.
2. Install JEI or REI.
3. Install KubeJS.
4. Create recipe like:
![image](https://github.com/GregTechCEu/GregTech-Modern/assets/52341158/7c50e1ae-faa1-4d85-ac04-9707de128ea2)
_Any recipe with a tag with at least 2 values is suitable for input._
5. Reload!
6. Try to use U button on terracotta.
7. Magic, no macerator recipe.

Important: But if you use R on wood dust, then crafting can be easily found in the list.

### Additional Information

I can't imagine why this is happening, I spent a lot of time debugging, but I never found out the reason.

Tap R on wood dust:
![image](https://github.com/GregTechCEu/GregTech-Modern/assets/52341158/5d79c736-6148-4cbd-9e4e-383f47bfda1d)

Tap U on Terracotta:
![image](https://github.com/GregTechCEu/GregTech-Modern/assets/52341158/474f015d-a268-451b-90c0-c4faf578b9c2)
_Recipe from gtceu, but not added by us. If you delete it, the result is the same._

**Comment by Exzept1on** [comment]
A little more debugging, I have now checked this recipe by creating it in my mod and the result is the same, it is not displayed if you press U on the input item, but it is displayed if you press R on the output item.

**Comment by Exzept1on** [comment]
![image](https://github.com/GregTechCEu/GregTech-Modern/assets/52341158/fbdbaa9c-0d4c-47ce-bd52-7980750cce88)
The already built-in recipe has the same problem.

**Comment by MindBrain27** [comment]
I had some issues like this and found a fix, when using #forge:dusts/ender_pearl as an input, no recipes were seen in REI, they did work though. Switching the input to a strict item, gtceu:ender_pearl_dust fixed it, it was an issue for every machine type i tried. Tags seem to be an issue for certain items, I do have AE2 and Almost Unified installed and ender pearl dusts being unified as well and I think it has something to do with that. Hope this helps.

**Comment by MindBrain27** [comment]
Do you have Almost Unified installed? I had issues with this and finally realized it was due to the fact that I had Ender IO installed but everything hidden except Conduits, every single one of the dusts from Ender IO caused this problem. When GT looks for a tagged item that is hidden, it breaks the REI/JEI display. If you have more than 1 Coal Dust and 1 is hidden, it will cause an issue. My fix was to remove the hidden items from the tags. After I removed all of Ender IO's dusts from the dust tags I saw all of the recipes come back.

TLDR; Hiding a (shared) tagged item will cause JEI/REI to not show recipes.

**Comment by Exzept1on** [comment]
Yeah, that shit was because of him, so I deleted it.
